### ISSUE TITLE
Y24-098 Tube Rack Banking Purpose

### ISSUE THREAD

**Issue by yoldas** [sanger/limber#1695]
**User story**
As PSD, we would like to support Limber plate purpose configuration using the versioned Tube Rack model proposed in Reuse of TubeRacks document.

**Who are the primary contacts for this story**
Abdullah, Katy, Andrew

**Who is the nominated tester for UAT**
TBD

**Acceptance criteria**
- [ ] LRC Bank Seq and LRC Bank Spare tube racks are configured in purpose config (they are tube purposes in DPL-811)

**Dependencies**
This story is blocked by the following dependencies:
- #1482

References
This story has a non-blocking relationship with:
- #1314
- #1713

**Additional context**
The versioned tube rack model is researched in DPL-1011 Re-use of tube racks and the model is summarised in the Google Document[ DPL-1011 Re-use of tube racks](https://docs.google.com/document/d/1rvPXeCk5rBEf72VkchHCD2FRmWANboke2pln_Lt_j5U/edit?usp=sharing)

This story is part of re-implementation of [DPL-811 Transfer PBMC isolations into FluidX tube racks](https://github.com/sanger/limber/issues/1314) (Banking) using the versioned tube rack model.

Story relationships are shown in the Lucidchart diagram [TubeRack Stories](https://lucid.app/lucidchart/d88bdcfd-9628-4fec-8b5a-453f680ee06b/edit?viewport_loc=-1051%2C-316%2C2770%2C1486%2C0_0&invitationId=inv_13d74286-1cc2-4c2f-aaf0-30887764601c) .

It is not possible to release the implementation of this story in isolation.

**Comment by TWJW-SANGER** [comment]
One more user facing acceptance criteria?:

- [ ] Tube racks are displayed and linked to on the Limber Pipeline Work In Progress page

**Comment by yoldas** [comment]
If we display TubeRacks in Work In Progress page, what happens?
For example, in the following page: https://uat.limber.psd.sanger.ac.uk/pipeline_work_in_progress/scRNA%20Core%20Cell%20Extraction

At the moment we are showing LRC Bank Seq  and LRC Bank Spare tubes in the last two column individually. We would instead show a single Tube Rack entry for its LRC Bank Seq tubes and similarly a single Tube Rack entry for its LRC Bank Spare tubes, which would take us to their TubeRack views similar to a Plate views. 

I think modifying Work In Progress view would deserve a separate story but I can still add it here for now.

**Comment by KatyTaylor** [comment]
I think it would be handy if you mentioned in your acceptance criteria which are existing already. 
e.g. a tube rack can already have a purpose, so you could put:
- Tube Rack can have a purpose (already implemented)

Or, just leave it out altogether?

What actually needs to be done here? Is the main thing "Pipeline configuration can specify relationships using tube racks"?

**Comment by yoldas** [comment]
I have added Purpose config and pipeline config criteria similar to DPL-811 in order to group Seq and Spare tubes under the tube racks with the same purpose.

**Comment by yoldas** [comment]
Move criterion "Tube racks are displayed and linked to on the Limber Pipeline Work In Progress page" to another story.

**Comment by yoldas** [comment]
I have moved the "Tube racks are displayed and linked to on the Limber Pipeline Work In Progress page" to a new story: Y24-112 Display Banking Tube Racks for scRNA Core Cell Extraction in Limber Pipeline Work in Progress #1714

I have made a top level story for the Banking part to link the banking stories together: Y24-088 Implement DPL-811 Transfer PBMC Isolations into FluidX tube racks using the versioned TubeRack model #1713
